blockchain/skipchain/mod.go

```diff
@@ -172,8 +172,34 @@ func (a skipchainActor) InitChain(data proto.Message, players mino.Players) erro
 		return xerrors.New("players must implement cosi.CollectiveAuthority")
 	}
 
+	_, err := a.db.Read(0)
+	if err == nil {
+		// Genesis block already exists.
+		return nil
+	}
+
+	if !xerrors.Is(err, NewNoBlockError(0)) {
+		return xerrors.Errorf("couldn't read the genesis block: %v", err)
+	}
+
 	conodes := newConodes(ca)
+	iter := conodes.AddressIterator()
+
+	if conodes.Len() > 0 && iter.GetNext().Equal(a.mino.GetAddress()) {
```

Comment:
what if `GetNext()` returns nil?

Reply:
That's I check the length before

Reply:
Oh, right, I didn't pay attention. 
In that case that would make more sense using `iter.HasNext()` instead of ` conodes.Len() > 0`

Reply:
Oops, yes. That's better.